android-json-form-wizard/src/main/java/com/vijay/jsonwizard/fragments/JsonFormFragment.java

```diff
@@ -299,12 +299,8 @@ public void skipLoadedStepsOnNextPressed() {
     private int getFormStepNumber(String nextFormNumber) {
         int formNumber = 0;
         if (StringUtils.isNotBlank(nextFormNumber)) {
-            int currentFormNumber = Integer.parseInt(nextFormNumber.substring(4, 5)) - 1;
-            if (currentFormNumber > 0) {
-                formNumber = currentFormNumber;
-            } else if (currentFormNumber == 0) {
-                formNumber = 1;
-            }
+            int currentFormNumber = Integer.parseInt(nextFormNumber.substring(4)) - 1;
```

Comment:
Why do we need to subtract 1?

Reply:
If we are trying to get the current step number, won't https://github.com/OpenSRP/opensrp-client-native-form/pull/407/files#diff-4af3b3a5f5179b96a2f8051da8ad980ec341e72fc18aa4bd23f0cbc2c2216845L739 suffice?

The current logic assumes that the next step will always be +1 which is not always certain to be true.